mlflow/ai_commands/genai/analyze_experiment.md

```diff
@@ -51,35 +58,36 @@ Analyzes traces in an MLflow experiment for quality issues, performance problems
   - If not, fall back to Options 1 or 2
 
 - Ask user for the path to their environment file (if using Options 1-2)
-- Verify connection by listing experiments: `uv run --env-file <env_file_path> python -m mlflow experiments search --max-results 10`
-- **Option to search by name**: If user knows the experiment name, use `--filter-string` parameter:
-  - `uv run --env-file <env_file_path> python -m mlflow experiments search --filter-string "name LIKE '%experiment_name%'" --max-results 10`
-- Ask user for experiment ID or let them choose from the list
-- **WAIT for user response** - do not continue until they provide the experiment ID
-- Add `MLFLOW_EXPERIMENT_ID=<experiment_id>` to their environment file
-- Run `uv run --env-file <env_file_path> python -m mlflow traces --help` to understand the CLI commands and options
+- Test connection: `uv run --env-file <env_file_path> python -m mlflow experiments search --max-results 5`
+
+### 1.2 Select Experiment and Test Trace Retrieval
+
+- If MLFLOW_EXPERIMENT_ID not already set, list available experiments and ask user to select one:
 
-### 1.2 Test Trace Retrieval
+  - **Option to search by name**: Use filter_string parameter for name search
+  - Ask user for experiment ID or let them choose from the list
+  - **WAIT for user response** - do not continue until they provide the experiment ID
+  - For CLI: Add `MLFLOW_EXPERIMENT_ID=<experiment_id>` to environment file
 
-- Call `uv run --env-file <env_file_path> python -m mlflow traces search --max-results 5` to verify:
+- Search for traces (max_results=5) to verify:
   - Traces exist in the experiment
-  - CLI is working properly (using local MLflow installation)
-  - Database connection is valid
+  - Tools are working properly
+  - Connection is valid
 - Extract sample trace IDs for testing
-- Get one full trace with `uv run --env-file <env_file_path> python -m mlflow traces get --trace-id <id>` to understand the data structure
+- Get one full trace by trace_id to understand the data structure
```

Comment:
Would it help to tell the bots to try to get one that isn't in ERROR state?

Reply:
good idea, yes! done :)